src/hooks/use-link-provider-tile-dialog.tsx

```diff
@@ -0,0 +1,99 @@
+import React, { useRef, useState } from "react";
+import LinkGraphIcon from "../clue/assets/icons/table/link-graph-icon.svg";
+import { useCustomModal } from "./use-custom-modal";
+import { ITileLinkMetadata } from "../models/tiles/tile-link-types";
+import { ITileModel } from "../models/tiles/tile-model";
+import { getTileContentById } from "../utilities/mst-utils";
+import { TableContentModelType } from "../models/tiles/table/table-content";
+
+import "./link-tile-dialog.scss";
+
+//TODO: use-link-table-dialog-dataflow.tsx is very similar
+//consider refactoring -> https://www.pivotaltracker.com/n/projects/2441242/stories/184992684
+
+const isLinkedToTile = (model: ITileModel, tileId: string) => {
+  const tileContent = getTileContentById(model, tileId) as TableContentModelType;
+  return tileContent?.linkedTiles?.includes(model.id);
+};
```

Comment:
This isn't quite right, given that it specifically assumes that it's a `TableContentModel` when it could be another provider, and it assumes the presence of a `linkedTiles` method. My inclination would be to put functions like this in a utilities module where they're more widely accessible. If you look at the `shared-data-utils.ts` module in CODAP you can see some examples. You might even be able to use some of those functions directly but others might be a bit different since CLUE generally links by tile id rather than data set id, but it should give the idea. Ultimately, it should be possible to decide whether one tile is linked to another without assuming the type of either tile or requiring that each tile implement a `linkedTiles` method.